tests/ihontaryk/homework_4/test_collections.py

```diff
@@ -0,0 +1,32 @@
+import pytest
+
+
+@pytest.mark.parametrize('data1, expected_result',
+                         [(
+                                 [{'username': 'yellowrabbit122', 'last_name': 'Marshall', 'first_name': 'Zoey',
+                                   'gender': 'female'},
+                                  {'username': 'ticklishdog243', 'last_name': 'Marshall', 'first_name': 'Herman',
+                                   'gender': 'male'},
+                                  {'username': 'silverfrog870', 'last_name': 'Robinson', 'first_name': 'Brian',
+                                   'gender': 'male'},
+                                  {'username': 'heavykoala856', 'last_name': 'Andrews', 'first_name': 'Liam',
+                                   'gender': 'male'},
+                                  {'username': 'goldenbird473', 'last_name': 'Andrews', 'first_name': 'Jacqueline',
+                                   'gender': 'female'}],
+                                 [{'username': 'goldenbird473', 'last_name': 'Andrews', 'first_name': 'Jacqueline',
+                                   'gender': 'female'},
+                                  {'username': 'heavykoala856', 'last_name': 'Andrews', 'first_name': 'Liam',
+                                   'gender': 'male'},
+                                  {'username': 'ticklishdog243', 'last_name': 'Marshall', 'first_name': 'Herman',
+                                   'gender': 'male'},
+                                  {'username': 'yellowrabbit122', 'last_name': 'Marshall', 'first_name': 'Zoey',
+                                   'gender': 'female'},
+                                  {'username': 'silverfrog870', 'last_name': 'Robinson', 'first_name': 'Brian',
+                                   'gender': 'male'}]
+                         )])
+def test_sorting_by_2_parameters(data1, expected_result):
+    """
+    verify sorting by last name and then first name
+    """
+
+    assert sorted(data1, key=lambda i: (i['last_name'], i['first_name'])) == expected_result
```

Comment:
to inspect expected result you can simply provide list  of usernames in order you expect
and during assert do:

```
res  =   sorted(data1, key=lambda i: (i['last_name'], i['first_name'])) 
assert [i["user_name"] for i in res] == expected_result



```

Reply:
try to avoid data not needed for test purposes

Reply:
I use such result on purpose to try tuples according to this homework study materials